src/parsers/manifestjson.js

```diff
@@ -22,16 +24,33 @@ export default class ManifestJSONParser {
 
     try {
       this.parsedJSON = JSON.parse(jsonString);
-    } catch (error) {
-      var errorData = {
-        code: 'MANIFEST_JSON_INVALID',
-        message: 'Invalid JSON in manifest file.',
-        file: 'manifest.json',
-        description: error,
-      };
-      this.collector.addError(errorData);
-      this.isValid = false;
-      return;
+    } catch (originalError) {
+      // First we'll try to remove comments with esprima;
+      // WebExtension manifests can contain comments, so we'll strip
+      // them out and see if we can parse the JSON.
+      // If not it's just garbage JSON and we error.
+      //
+      // Originally from https://github.com/abarreir/crx2ff/blob/d2b882056f902d751ad05e329efda7eddcb9d268/libs/ext-converter.js#L19-L37
+      var manifestString = `var o = ${jsonString}`;
+      try {
+        var tokens = esprima.tokenize(manifestString).slice(3);
+        var cleanedJSONString = tokens.reduce((json, token) => {
+          return `${json}${token.value}`;
+        }, '');
```

Comment:
We should totally document what the hell this does, I looked up `esprima` and wondered that this is an actual JavaScript parser, then noticed, yeah, JSON is probably just a subset of JavaScript so it'll work, parse it and then you're fetching out only relevant tokens.

On the other hand I also found 

 * http://stackoverflow.com/a/23753148
 * and https://github.com/judofyr/timeless/issues/57#issuecomment-31872462

I don't know about ES6 but it appears there are very very minimal differences. We should test them and at least expect them to fail or make sure that they succeed by some magic.